js/sign/tests/deterministic_test.js

```diff
@@ -0,0 +1,153 @@
+import * as det from '../lib/cbor/deterministic.js';
+import * as ainfo from '../lib/cbor/additionalinfo.js';
+
+const uInts = {
+  0: new Uint8Array([0x00]),
+  10: new Uint8Array([0x0a]),
+  23: new Uint8Array([0x17]),
+  24: new Uint8Array([0x18, 0x18]),
+  45: new Uint8Array([0x18, 0x2d]),
+  255: new Uint8Array([0x18, 0xff]),
+  256: new Uint8Array([0x19, 0x01, 0x00]),
+  5000: new Uint8Array([0x19, 0x13, 0x88]),
+  65535: new Uint8Array([0x19, 0xff, 0xff]),
+  65536: new Uint8Array([0x1a, 0x00, 0x01, 0x00, 0x00]),
+  4294967: new Uint8Array([0x1a, 0x00, 0x41, 0x89, 0x37]),
+  4294967295: new Uint8Array([0x1a, 0xff, 0xff, 0xff, 0xff]),
+  // This is also the max length of Uint8Array on node.
+  4294967296: new Uint8Array([
+    0x1b, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+  ]),
+  // Number.MAX_SAFE_INTEGER
+  9007199254740991: new Uint8Array([
+    0x1b, 0x00, 0x1f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+  ]),
+  '9223372036854775807': new Uint8Array([
+    0x1b, 0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+  ]),
+  // Max CBOR supported value 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF.
+  '18446744073709551615': new Uint8Array([
+    0x1b, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+  ]),
+};
+
+describe('Deterministic check', () => {
```

Comment:
I'll split them based on the type being tested instead (uint, text, bytestring, array, map, additional info etc), because checkDeterministic will be run for everything so it would then bloat.